client.js에서`const client = axios.create()`라고 정의 이후

auth.js에서
```js
export const login = ({username, password}) => client.post('/api/auth/login', {username, password});
```
이 auth라는 모듈은 redux-saga에서 관리
```js
export default function createRequestSaga(type, request){
    const SUCCESS = `${type}_SUCCESS`
    const FAILURE = `${type}_FAILURE`   

    return function *(action){
        yield put(startLoading(type))
        try{
            const response = yield call(request, action.payload)
            yield put({type : SUCCESS, payload : response.data})
        }catch(e){ 
            yield put({type : FAILURE, payload : e, error : true})
        }
        yield put(finishLoading(type))
    }

import createRequestSaga, {createRequestActionTypes}  from '../lib/createRequestSaga'
const [LOGIN, LOGIN_SUCCESS, LOGIN_FAILURE] = createRequestActionTypes('auth/LOGIN') 
export const login = createAction(LOGIN, ({ username, password }) =>({ username, password })) 
const loginSaga = createRequestSaga(LOGIN, authAPI.login);
export function * authSaga(){
    yield takeLatest(REGISTER, registerSaga);
    yield takeLatest(LOGIN, loginSaga);
}
const auth = handleActions(
    {
        [LOGIN_SUCCESS] : (state, {payload : auth}) => ({...state, authError : null, auth}), 
        [LOGIN_FAILURE] : (state, {payload : error}) => ({...state, authError : error})
    }, 
    initialState
) 
```
위의 코드처럼 성공과 실패를 나눠서 authError를 null로 하거나 error를 담음.

만약 서버에서 데이타를 보내줄 때 status를 200으로 설정하면 자연스레 성공이 발동 > 잘처리가 됨. 
하지만 204로 맞춘다면? 
status를 204로 보낸다면 작동이 아예 안됨. 

authError는 null, auth는 아예 "" 

200으로 맞춘다면? 
null {success: true, message: "성공적으로 토큰이 발급되며 로그인이 되었습니다.", username: "홍철이"}
이런식으로 됨. 

400으로 맞춘다면? 

null undefined

이렇게 됨. 

서버코드는 다음과 같음.
```js
exports.login = wrapE(async(req, res, next) => { 
    const {username, password} = req.body;  
    await schema.validateAsync({username, password}) 
    //스키마에 맞는지를 확인하는 검증 로직 필요 
    const user = await User.findOne({"username" : username}).lean(); 
    if(!user){
        throw new Error(`${username}님은 존재하지 않는 사용자입니다.`); 
    } 
    const hashedPW = user.password;
    const isOwn = bcrypt.compareSync(password, hashedPW); 

    if(isOwn){
        await generateToken(res, user._id, username)
        res.status(400).send({
          success: true,
          message: '성공적으로 토큰이 발급되며 로그인이 되었습니다.', 
          username : username
        });
        return;
    }else{
        throw new Error(`잘못된 비밀번호를 입력하셨습니다.`); 
    }
}) 
```
질문 redux-saga에서 http status코드를 보고 success와 failure를 판단하는 건가요? 
제 코드내에서 status code에 관해서 짠 로직 자체가 없는데 어떻게 관리가 되나요?